e2e-tests-rust/tests/lib.rs

```diff
@@ -333,3 +334,15 @@ async fn cant_load_into_existing_state() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn set_chain_id() -> anyhow::Result<()> {
+    let provider = init_testing_provider(|node| node.block_time(3)).await?;
+
+    let new_chain_id = 261;
+    provider.anvil_set_chain_id(new_chain_id).await?;
+
+    assert_eq!(new_chain_id, provider.get_chain_id().await?);
+
```

Comment:
To be honest with you I am still not sure what the semantics here should be like after changing chain id. But if it makes sense to do so, maybe you could assert that we can still finalize transactions here? Something like this:

```suggestion
    // Assert we can still finalize transactions after changing chain id
    provider.tx().finalize().await?;
```

Reply:
Makes sense, added test to ensure transactions can be executed after the chain id change.